waiter/src/waiter/auth/oidc.clj

```diff
@@ -328,20 +358,28 @@
    request-handler]
   (let [oidc-authority (utils/uri-string->host oidc-authorize-uri)]
     (fn oidc-auth-handler [request]
-      (let [oidc-mode-delay (delay (request->oidc-mode allow-oidc-auth-api? allow-oidc-auth-services? oidc-default-mode request))]
-        (cond
-          (or (auth/request-authenticated? request)
-              (= :disabled @oidc-mode-delay)
-              ;; OIDC auth is no-op when request cannot be redirected
-              (not (supports-redirect? oidc-authority oidc-redirect-user-agent-products request))
-              ;; OIDC auth is avoided if client already has too many challenge cookies
-              (too-many-oidc-challenge-cookies? request oidc-num-challenge-cookies-allowed-in-request))
-          (request-handler request)
+      (-> (let [oidc-mode-delay (delay (request->oidc-mode allow-oidc-auth-api? allow-oidc-auth-services? oidc-default-mode request))
+                skip-oidc? (or (auth/request-authenticated? request)
+                               (= :disabled @oidc-mode-delay)
+                               ;; OIDC auth is no-op when request cannot be redirected
+                               (not (supports-redirect? oidc-authority oidc-redirect-user-agent-products request)))
+                many-oidc-challenge-cookies? (delay (too-many-oidc-challenge-cookies? request oidc-num-challenge-cookies-allowed-in-request))]
+            (cond
+              (or skip-oidc?
+                  ;; OIDC auth is avoided if client already has too many challenge cookies
+                  @many-oidc-challenge-cookies?)
+              (cond-> (request-handler request)
+                ;; update a 401 response to 403 if there were too many cookies and downstream auth options are disabled
+                (not skip-oidc?)
+                (ru/update-response
+                  (make-oidc-auth-too-many-cookies-response-updater request oidc-num-challenge-cookies-allowed-in-request)))
 
-          :else
-          (ru/update-response
-            (request-handler request)
-            (make-oidc-auth-response-updater jwt-auth-server @oidc-mode-delay oidc-same-site-attribute password request)))))))
+              :else
+              (ru/update-response
+                (request-handler request)
+                (make-oidc-auth-response-updater jwt-auth-server @oidc-mode-delay oidc-same-site-attribute password request))))
+          ;; do not let the downstream waiter/auth-disabled? on the response flow out of the OIDC auth handler
```

Comment:
I believe that "comments should describe *why* rather than *what*" is a good philosophy. Code should be self-documenting as to _what_ it's doing, as I think is the case here. What is unclear to me is _why_ we don't want this property to flow out of the OIDC auth handler.

Reply:
Improved the comment with why we want it removed - to remove confusion in debugging in other handlers.